scripts/isWorkspace.js

```diff
@@ -0,0 +1,22 @@
+import path from "node:path";
+import {fileURLToPath} from "node:url";
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+// Using CommonsJS require.resolve as long as import.meta.resolve is experimental
+import {createRequire} from "node:module";
+const require = createRequire(import.meta.url);
+
+async function main() {
+	const cPath = require.resolve("@ui5/cli/package.json");
+	const cliPath = path.relative(__dirname, path.dirname(cPath));
+
+	// no workspace detected
+	if (!cliPath.startsWith("..")) {
+		process.exit(1);
+	}
+}
+
+main().catch((error) => {
+	console.log(error);
```

Comment:
This won't be executed as "main" already calls process.exit.

AFAIK in ES modules you can also use async/await in the root context, so you don't need to define a function at all.

Reply:
Refactored in 1c310e2.